examples/systemcalls/systemcalls.c

```diff
@@ -38,62 +50,149 @@ bool do_exec(int count, ...)
+        va_end(args);
+        return false;
     }
-    command[count] = NULL;
-    // this line is to avoid a compile warning before your implementation is complete
-    // and may be removed
-    command[count] = command[count];
-
-/*
- * TODO:
- *   Execute a system command by calling fork, execv(),
- *   and wait instead of system (see LSP page 161).
- *   Use the command[0] as the full path to the command to execute
- *   (first argument to execv), and use the remaining arguments
- *   as second argument to the execv() command.
- *
-*/
 
-    va_end(args);
+    // Create an array to hold the command and its arguments
+    char* command[count + 1]; // +1 for the NULL terminator
+    for (int i = 0; i < count; i++) 
+    {
+        command[i] = va_arg(args, char*);
+    }
+    command[count] = '\0'; // Null-terminate the array
+
+    // Fork a new process
+    pid_t child_pid = fork();
+
+    if (child_pid < 0) 
+    {
+        // Forking failed
+        va_end(args);
+        return false;
+    } 
+    else if (child_pid == 0) 
+    {
+        // This is the child process
+        // Execute the command in the child process
+        execv(command[0], command);
+
+        // If execv() returns, it means an error occurred
+        // Print an error message and exit with failure status
+        perror("execv");
+        exit(EXIT_FAILURE);
+    } 
+    else 
+    {
+        // This is the parent process
+        int status;
+        // Wait for the child process to complete
+        waitpid(child_pid, &status, 0);
+
+        va_end(args);
 
-    return true;
+        // Check if the child process exited successfully
+        if (WIFEXITED(status) && WEXITSTATUS(status) == 0) 
+        {
+            return true;
+        } 
+        else 
+        {
+            return false;
+        }
+    }
 }
 
+
 /**
 * @param outputfile - The full path to the file to write with command output.
 *   This file will be closed at completion of the function call.
 * All other parameters, see do_exec above
 */
+//Credits : Asked GPT to generate the code for this function, the prompt was same as the TODO.
 bool do_exec_redirect(const char *outputfile, int count, ...)
 {
     va_list args;
     va_start(args, count);
-    char * command[count+1];
-    int i;
-    for(i=0; i<count; i++)
+
+    // Check if the count is valid
+    if (count <= 0) 
     {
-        command[i] = va_arg(args, char *);
+        va_end(args);
+        return false;
     }
-    command[count] = NULL;
-    // this line is to avoid a compile warning before your implementation is complete
-    // and may be removed
-    command[count] = command[count];
-
-
-/*
- * TODO
- *   Call execv, but first using https://stackoverflow.com/a/13784315/1446624 as a refernce,
- *   redirect standard out to a file specified by outputfile.
- *   The rest of the behaviour is same as do_exec()
- *
-*/
 
-    va_end(args);
+    // Create an array to hold the command and its arguments
```

Comment:
@Rishikesh38 think about how you could avoid duplicating the code above with this function.